### PR TITLE
Module-3 -BaseTest

### PR DESCRIPTION
**Pull request by Linases**
I am not sure... I created two BaseTest classes by logic, but it can be one as well. 
image

### CODE REVIEW

File: Functionality_Tests/FunctionalityTests.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+using System.Threading.Tasks;
+using OpenQA.Selenium.Chrome;
+using OpenQA.Selenium.Support.UI;
+using SeleniumExtras.WaitHelpers;
+using OpenQA.Selenium.DevTools.V115.Page;
+
+namespace Functionality_Tests_Suit
+{
+    [TestFixture]
+
```
Comment (Alina150601) [inline Functionality_Tests/FunctionalityTests.cs]: blank line :) review your code for the rule
There should be blank lines between all methods, no blank lines between attribute and method

File: Functionality_Tests/FunctionalityTests.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            Assert.That(elementItemsList, Is.Ordered.Ascending);
+        }
+    
+        [Test, Order(2)]
+
+        public void RemoveFromCart()
+        {
+            AddProductToCart();
+            var elementShopingcartIcon = driver.FindElement(By.ClassName("shopping_cart_link"));
+            elementShopingcartIcon.Click();
```
Comment (Alina150601) [inline Functionality_Tests/FunctionalityTests.cs]: Please add check that after clicking the "Add to Cart" button for the selected product the cart icon in the top-right corner displays the correct number of items added (e.g., 1 item).

File: Functionality_Tests/FunctionalityTests.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            var elementRemoveButton = driver.FindElement(By.XPath("//button[text() = 'Remove']"));
+            elementRemoveButton.Click();
+            var elementShopingcartIcon2 = driver.FindElement(By.ClassName("shopping_cart_link"));
+            Assert.That(elementShopingcartIcon2.Text.Contains('1'), Is.False);
```
Comment (Alina150601) [inline Functionality_Tests/FunctionalityTests.cs]: But what if, after deleting the item from the cart, the count of products will be not 0, but 2?) test will pass, but shouldn't.
You can add assert that the element with number is not displayed instead check that number 1 is not present.
image

File: Functionality_Tests/FunctionalityTests.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        }
+        [Test, Order(3)]
+
+        public void Logout()
+        {
+            SuccessfulLogin();
+            var elementMeniuButton = driver.FindElement(By.Id("react-burger-menu-btn"));
+            elementMeniuButton.Click();
+            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(20));
```
Comment (Alina150601) [inline Functionality_Tests/FunctionalityTests.cs]: use var :) and below

File: Functionality_Tests/FunctionalityTests.cs (hunk 2 above)
Comment (Linases) [reply Functionality_Tests/FunctionalityTests.cs]: fixed

File: Functionality_Tests/FunctionalityTests.cs (hunk 1 above)
Comment (Linases) [reply Functionality_Tests/FunctionalityTests.cs]: fixed

File: Functionality_Tests/FunctionalityTests.cs (hunk 3 above)
Comment (Linases) [reply Functionality_Tests/FunctionalityTests.cs]: fixed, thank you for the useful help!

File: Functionality_Tests/FunctionalityTests.cs (hunk 4 above)
Comment (Linases) [reply Functionality_Tests/FunctionalityTests.cs]: fixed

File: Functionality_Tests/BaseTest.cs [hunk 5]
```diff
@@ -10,7 +10,7 @@
 
 namespace Functionality_Tests_Suit
 {
-    public abstract class BaseTest
+    public class BaseTest
     {
         public IWebDriver driver;
```
Comment (Alina150601) [inline Functionality_Tests/BaseTest.cs]: Should be private and readonly. Rename according to codestyles after it :)
Name of fields:
private fields _lowerCaseWithUnderScore 
public or protected fields UpperCase

File: Functionality_Tests/BaseTest.cs [hunk 6]
```diff
@@ -10,7 +10,7 @@
...
         public string mainUrl;
```
Comment (Alina150601) [inline Functionality_Tests/BaseTest.cs]: Should be readonly, also you can use it in SeleniumTests and if so, it should be protected

Comment (Linases) [reply Functionality_Tests/BaseTest.cs]: fixed

File: Functionality_Tests/BaseTest.cs (hunk 5 above)
Comment (Linases) [reply Functionality_Tests/BaseTest.cs]: Done protected as well since it is used in SeleniumTests also

Comment (Alina150601) [reply Functionality_Tests/BaseTest.cs]: exactly, great!

**Pull request merged**